### PR TITLE
feat: add stBadOpcode

### PR DESCRIPTION
**Pull request by Eikix**
Time spent on this PR:

Begins work on #89 

# Pull Request type

Please check the type of change your PR introduces:

-   [ ] Bugfix
-   [x] Feature
-   [ ] Code style update (formatting, renaming)
-   [ ] Refactoring (no functional changes, no API changes)
-   [ ] Build-related changes
-   [ ] Documentation content changes
-   [x] Testing

# What is the new behavior?

# Does this introduce a breaking change?

-   [ ] Yes
-   [x] No

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -30,8 +30,8 @@ revm-primitives = "1.1"
 reth-rlp = { git = "https://github.com/paradigmxyz/reth.git", tag = "v0.1.0-alpha.7" }
 
 # Kakarot deps
-kakarot-rpc-core = { git = "https://github.com/kkrt-labs/kakarot-rpc.git", rev = "0894457" }
-kakarot-test-utils = { git = "https://github.com/kkrt-labs/kakarot-rpc.git", rev = "0894457" }
+kakarot-rpc-core = { git = "https://github.com/kkrt-labs/kakarot-rpc.git", rev = "ae5e220" }
```
Comment (jobez) [inline Cargo.toml:33]: this is the bump for steps right?

File: crates/ef-testing/src/models/case.rs [hunk 2]
```diff
@@ -107,7 +109,14 @@ impl BlockchainTestCase {
...
                 | "OutOfGasContractCreation.json" // ef-test #116
                 | "CallContractToCreateContractAndCallItOOG.json" // ef-test #117
-        )
+                | "undefinedOpcodeFirstByte.json" // ef-tests #121
+                | "measureGas.json" // ef-tests #122
+                | "badOpcodes.json" // ef-tests #123
+                | "operationDiffGas.json" // ef-tests #124
+                | "invalidDiffPlaces.json" // ef-tests #125
+                | "invalidAddr.json" // ef-tests #126
+        ) || matches!(name, name if name.starts_with("opc") && name.ends_with(".json"))
```
Comment (jobez) [inline crates/ef-testing/src/models/case.rs:118]: maybe we can create an issue about having two distinct places to add skips, by specific case or by suite.

**Review by jobez — CHANGES_REQUESTED** [review]

File: crates/ef-testing/src/models/case.rs (hunk 2 above)
Comment (Eikix) [reply crates/ef-testing/src/models/case.rs:118]: wdym

File: Cargo.toml (hunk 1 above)
Comment (Eikix) [reply Cargo.toml:33]: I don't think so, I think it was a small bump by me when I thought it changed something (it doesn't)

File: crates/ef-testing/src/models/case.rs (hunk 2 above)
Comment (Eikix) [reply crates/ef-testing/src/models/case.rs:118]: ohh skipping a whole suite?

Comment (jobez) [reply crates/ef-testing/src/models/case.rs:118]: kk, talked with greg and it seems we arent distinguishing case/suite rn for skips (case is a specific key/val pair within a json file, json file is a suite). what i meant was we'd have a `skipped_by_name` and a `skipped_by_regex` so its less of a scary big list to manage.

Comment (Eikix) [reply crates/ef-testing/src/models/case.rs:118]: ah i see

so case: key/val pair within a json
suite: json file
folder of tests: ?

Comment (Eikix) [reply crates/ef-testing/src/models/case.rs:118]: you're right! let's create the issue and improve clarity

File: Makefile [hunk 3]
```diff
@@ -20,10 +20,10 @@ KAKAROT_COMMIT := .katana/remote_kakarot_sha
 .PHONY: $(EF_TESTS_DIR)
 setup: $(EF_TESTS_DIR)
 
-fetch-dump:
+fetch-dump: fetch-kakarot-submodule-commit
```
Comment (jobez) [inline Makefile:23]: this was changed because you had issues with loading the latest commit's dump, oui?

**Review by jobez — APPROVED** [review]
left a comment guessing why the makefile changes were included in this, other than that lgtm

File: Makefile (hunk 3 above)
Comment (Eikix) [reply Makefile:23]: yes, thank you
problem was that if the fetch already existed, it wouldnt fetch once more

**Pull request merged**